### ISSUE TITLE
for(a in array) iterates over the entire enumerable attributes in Array.prototype

### ISSUE THREAD

**Issue by HugoCrd** [knalli/angular-vertxbus#212]
### Subject of the issue
Delegator uses `for (let address in this.handlers)` to iterate over callbacks
https://github.com/knalli/angular-vertxbus/blob/a3445fdc3f7c33644c1a447fc85872cf7254d060/src/lib/service/Delegator.js#L15

If user has defined a new function in prototype with `Array.prototype.something = ...` this results in a bug. That is because `something` in this case is `enumerable` and that `for in` iterates over all enumerable attributes in prototype. 

### Your environment
Global issue

### Steps to reproduce
Add a function to Array prototype : 
```javascript
Array.prototype.something = function(){...}
```
Use angular-vertxbus.
Wait for `vertx-eventbus.system.connected` to happen

### Expected behaviour
Angular-vertxbus is ok and working great. Not even noticing `something` is in prototype :)

### Actual behaviour
`vertx-eventbus.system.connected` will never be fired because a exception has been fired here https://github.com/knalli/angular-vertxbus/blob/a3445fdc3f7c33644c1a447fc85872cf7254d060/src/lib/service/Delegator.js#L18

### Solutions

- https://github.com/knalli/angular-vertxbus/blob/a3445fdc3f7c33644c1a447fc85872cf7254d060/src/lib/service/Delegator.js#L16 : We should add : 
```javascript
if (!this.handlers.hasOwnProperty(address)) {
    return;
 }
```

- Workaround for users is to define correctly these functions using 
```javascript
Object.defineProperty(Array.prototype, "includes", {
    enumerable: false,
    value: function() {...}
})
```
But sometimes not doable...

**Comment by knalli** [comment]
Hm.. I thought that is not required any more running ES6 (because of `of`) or running transpiled into-ES5 code. 

Well, in that case the fix is probably better: `if ({}.hasOwnProperty.call(this.handlers, address))`

You can this a try, also making a PR. Thank you.

**Comment by knalli** [comment]
Fixed in 6.1.1. 

Actually it was wrong  `handlers` has been initialized with an array, it is actually a map (so: object) of arrays. But the same issue could be happen then also.

Said this, an hasOwnProperty check has been added. Thank you.
